Enable threading for XNNPACK ops.

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34547 Enable threading for XNNPACK ops.**
* #34319 Pass to remove prepacking ops.
* #34048 JIT pass to insert XNNPACK ops
* #34047 Integrate XNNPACK with custom class for packing weights.

Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

kimishpatel committed:
Enable threading for XNNPACK ops.

Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

AshkanAliabadi reviewed (approved):
Thanks

dreiss commented on cmake/Dependencies.cmake:
```diff
@@ -434,6 +434,7 @@ if(USE_XNNPACK)
       "${CONFU_DEPENDENCIES_BINARY_DIR}/XNNPACK")
 
     set_property(TARGET XNNPACK PROPERTY POSITION_INDEPENDENT_CODE ON)
+    target_compile_definitions(XNNPACK PRIVATE -Dpthreadpool_get_threads_count=pthreadpool_get_threads_count_xnnpack)
```
Please leave a comment explaining this hack.

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable threading for XNNPACK ops."


Summary:
This enables threading by passing a threadpool to xnnpack ops.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20370553](https://our.internmc.facebook.com/intern/diff/D20370553)

[ghstack-poisoned]
